src/platform_impl/macos/app_delegate.rs

```diff
@@ -36,10 +48,28 @@ lazy_static! {
             activation_hack::mouse_moved as extern "C" fn(&Object, Sel, id),
         );
 
+        decl.add_method(
+            sel!(application:openFile:),
+            application_open_file as extern "C" fn(&Object, Sel, id, id) -> BOOL,
+        );
+        decl.add_method(
+            sel!(application:openFiles:),
+            application_open_files as extern "C" fn(&Object, Sel, id, id),
+        );
+
         AppDelegateClass(decl.register())
     };
 }
 
+/// Copies the contents of the ns string into a `String` which gets returned.
+fn ns_string_to_rust(ns_string: id) -> String {
```

Comment:
```suggestion
unsafe fn ns_string_to_rust(ns_string: id) -> String {
```
I feel marking the function itself as `unsafe` better communicates that this function is unsafe to call/use (versus being safe to use but containing unsafe implementation details)

Now, you might've noticed that this module is actually full of functions that have an inner `unsafe` block, seemingly contradicting my advice! That's because marking those `extern` functions as `unsafe` would change their signature, which AFAIK wouldn't play nice with `add_method`.

Reply:
> I feel marking the function itself as unsafe better communicates that this function is unsafe to call/use

You are completely right

> marking those `extern` functions as `unsafe` would change their signature

Oh I had no idea about this. A quick internet search didn't yield any information; where could I learn more about this?

Reply:
> where could I learn more about this?

I don't really know where you can find more about this, but marking one of those `extern` functions as `unsafe extern` causes it to no longer type-check when passed to `add_method`.